### PR TITLE
fix: swap to yarn for thelounge instances

### PR DESCRIPTION
**Pull request by brettpetch**
## Fixes issues: 
- Un-tracked issue... https://github.com/thelounge/thelounge/issues/4532
- https://github.com/swizzin/swizzin/issues/856

## Proposed Changes:
- swap to thelounge
- add yarn installation to the npm source file

## Change Categories

- Bug fix 
- Change in `functions` 

## Checklist

- [x] Branch was made off the `develop` branch and the PR is targetting the `develop` branch
- [x] Docs have been made OR are not necessary
    - PR link: 
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Code conforms to project structure [(See more)](https://swizzin.ltd/dev/structure)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Testing was done
   - [x] Tests created or no new tests necessary
   - [x] Tests executed

### Architectures

|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|			|				|
| Bionic		|			|			|			|				|
| Buster		|			|			|			|				|
| Bullseye		|	✅		|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|	⚫️		|			|	⚫️		|	⚫️			|

### CODE REVIEW

**Review by Troublesome123 — APPROVED** [review]

**Review by Troublesome123 — APPROVED** [review]

File: scripts/update/lounge.sh [hunk 1]
```diff
@@ -6,7 +6,10 @@ function _uplounge() {
     if [[ $active == "active" ]]; then
         systemctl stop lounge
     fi
-    npm install -g thelounge@latest >> /dev/null 2>&1
+    npm uninstall -g thelounge --save >> /dev/null 2>&1
+    yarn_install
```
Comment (liaralabs) [inline scripts/update/lounge.sh:10]: Noisy

File: scripts/upgrade/lounge.sh [hunk 2]
```diff
@@ -12,7 +33,11 @@ if [[ $(systemctl is-active lounge) == "active" ]]; then
     echo_progress_done
 fi
 
-if ! npm update -g thelounge >> "$log"; then
+if thelounge --version lt 4.3.1; then
+    _yarnlounge
+fi
+
+if ! /usr/bin/yarn --non-interactive global upgrade thelounge >> "$log"; then
     echo_error "Lounge failed to update, please investigate the logs"
```
Comment (liaralabs) [inline scripts/upgrade/lounge.sh:41]: We should hard exit

File: scripts/upgrade/lounge.sh [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
 
+function _yarnlounge() {
+    #shellcheck source=sources/functions/npm
+    . /etc/swizzin/sources/functions/npm
+
+    if [[ $active == "active" ]]; then
+        systemctl stop lounge
+    fi
+    npm uninstall -g thelounge --save >> /dev/null 2>&1
+    yarn_install
```
Comment (liaralabs) [inline scripts/upgrade/lounge.sh:11]: Something is making noise here, guessing it's this function

**Comment by liaralabs** [comment]
image

Seems noisier than before? :thinking:

**Review by liaralabs — APPROVED** [review]

**Comment by liaralabs** [comment]
I fixed the test, but now the next yarn command on line 40 fails on a fresh test because yarn isn't getting installed due to lounge version being 4.3.1 already.

**Comment by liaralabs** [comment]
Perhaps the test should be re-written to check if it's installed via npm?

```
if npm list -g | grep -q lounge; then 
...
```

Quick test shows the following output after running through the install commands manually, so I think this might be preferable to version checking:
```
root@ubuntu1804:/etc/swizzin# npm list -g
/usr/lib
├── corepack@0.10.0
├── npm@8.5.5
├── sqlite3@
└── yarn@1.22.18

root@ubuntu1804:/etc/swizzin# yarn global list
yarn global v1.22.18
info "thelounge@4.3.1" has binaries:
   - thelounge
Done in 0.20s.
```

**Comment by brettpetch** [comment]
@liaralabs are we happy with how this is now? If so can we merge?

/b

File: scripts/remove/lounge.sh [hunk 4]
```diff
@@ -3,10 +3,12 @@
 systemctl disable -q lounge >> /dev/null 2>&1
 systemctl stop -q lounge
 
-npm uninstall -g thelounge --save >> /dev/null 2>&1
```
Comment (liaralabs) [inline scripts/remove/lounge.sh:6]: Wonder if we should just keep this? In case someone never runs upgrade but decided to remove the lounge? 🤔

Comment (brettpetch) [reply scripts/remove/lounge.sh:6]: Yeah, that seems smort. Will add it back.

**Pull request merged**